x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/support/Aggregations.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.featureindexbuilder.support;
+
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class Aggregations {
+    private Aggregations() {}
+
+    /**
+     * Supported aggregation by dataframe and corresponding meta information.
+     *
+     * aggregationType - the name of the aggregation as returned by
+     * {@link org.elasticsearch.search.aggregations.BaseAggregationBuilder#getType()}}
+     *
+     * targetMapping - the field type for the output, if null, the source type should be used
+     *
+     */
+    enum AggregationType {
+        AVG("avg", "double"),
+        MAX("max", null);
+
+        private final String aggregationType;
+        private final String targetMapping;
+
+        AggregationType(String name, String targetMapping) {
+            this.aggregationType = name;
+            this.targetMapping = targetMapping;
+        }
+
+        public String getName() {
+            return aggregationType;
+        }
+
+        public String getTargetMapping() {
+            return targetMapping;
+        }
+    }
+
+    private static Set<String> aggregationSupported = new HashSet<>(
+            Stream.of(AggregationType.values()).map(AggregationType::name).collect(Collectors.toList()));
```

Comment:
You should just use [Collectors.toSet()](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toSet--)
```
Stream.of(AggregationType.values()).map(AggregationType::name).collect(Collectors.toSet());
```